src/tools/cephfs/first-damage.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+# Ceph - scalable distributed file system
+#
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 2.1, as published by the Free Software
+# Foundation.  See file COPYING.
+
+# Suggested recovery sequence (for single MDS cluster):
+#
+# 1) Unmount all clients.
+#
+# 2) Flush the journal (if possible):
+#
+#    ceph tell mds.<fs_name>:0
+#
+# 3) Fail the file system:
+#
+#    ceph fs fail <fs_name>
+#
+# 4) Reset the journal:
```

Comment:
I’m not quite sure what context these suggested steps are in, but suggesting a journal reset without recover_dentries seems bad?

Reply:
My concern is that there may be metadata in an expired segment (untrimmed) which will poison the cache.

Reply:
`recover_dentries` is not necessary if (a) clients were unmounted and (b) the MDS flushed its journal before going offline.

Reply:
Let's put that nuance here then — recover_dentries if you have not done so since client operations ran, and then reset the journal.

Reply:
done.